source/server/data_moniker_service.cpp

```diff
@@ -240,10 +248,12 @@ Status DataMonikerService::StreamRead(ServerContext* context, const MonikerList*
 Status DataMonikerService::StreamWrite(ServerContext* context, ServerReaderWriter<StreamWriteResponse, MonikerWriteRequest>* stream)
 {
 #ifndef _WIN32
-  cpu_set_t cpuSet;
-  CPU_ZERO(&cpuSet);
-  CPU_SET(1, &cpuSet);
-  sched_setaffinity(0, sizeof(cpu_set_t), &cpuSet);
+  if(c_CpuAffinityConfig.stream_write >= 0) {
+    cpu_set_t cpuSet;
```

Comment:
Nit: this code is duplicated a few times. Consider a helper method or something like that so it's easier to tweak.

Reply:
Created a helper as suggested.